### PR TITLE
Add session parameter to get_available_datasets

### PR DESCRIPTION
**Pull request by femtotrader**
Should help https://github.com/pydata/pandas-datareader/issues/48

### CODE REVIEW

File: pandas_datareader/famafrench.py [hunk 1]
```diff
@@ -6,27 +6,29 @@
...
+    session = _init_session(session)
     try:
-        from lxml.html import parse
+        from lxml.html import document_fromstring
     except ImportError:
         raise ImportError("Please install lxml if you want to use the "
                           "get_datasets_famafrench function")
 
-    root = parse(_URL + 'data_library.html')
+    response = requests.get(_URL + 'data_library.html')
```
Comment (davidastephens) [inline pandas_datareader/famafrench.py]: Don't you have to do session.get(_URL...) in order to use the session?

Comment (femtotrader) [inline pandas_datareader/famafrench.py]: Yes you are right... sorry!

File: pandas_datareader/_utils.py [hunk 2]
```diff
@@ -26,4 +26,9 @@ class RemoteDataError(PandasError, IOError):
 if PANDAS_VERSION >= LooseVersion('0.14.0'):
     PANDAS_0140 = True
 else:
-    PANDAS_0140 = False
\ No newline at end of file
+    PANDAS_0140 = False
+
+def _init_session(session):
```
Comment (sinhrks) [inline pandas_datareader/_utils.py]: Should use the logic as current ``_BaseReader``, and ``_BaseReader`` can use this definition.

https://github.com/pydata/pandas-datareader/blob/master/pandas_datareader/base.py#L57

**Comment by femtotrader** [comment]
Yes I should do something like you did in WordBankReader

https://github.com/pydata/pandas-datareader/blob/master/pandas_datareader/wb.py

**Pull request closed without merging**